src/main/java/bio/terra/service/filedata/flight/ingest/IngestCopyLoadHistoryToBQStep.java

```diff
@@ -1,92 +1,80 @@
 package bio.terra.service.filedata.flight.ingest;
 
-import bio.terra.model.BulkLoadHistoryModel;
 import bio.terra.service.dataset.Dataset;
 import bio.terra.service.dataset.DatasetService;
 import bio.terra.service.load.LoadService;
-import bio.terra.service.load.flight.LoadMapKeys;
 import bio.terra.service.tabulardata.google.BigQueryPdao;
-import bio.terra.stairway.*;
-import java.time.Instant;
-import java.util.List;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
 import java.util.UUID;
 import java.util.concurrent.TimeUnit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class IngestCopyLoadHistoryToBQStep implements Step {
-  private static final Logger logger = LoggerFactory.getLogger(IngestBulkFileResponseStep.class);
+public class IngestCopyLoadHistoryToBQStep extends IngestCopyLoadHistoryStep {
```

Comment:
This is an interesting pattern - and something that we're running into more and more as we handle GCP vs. Azure workflows. 

With the pattern you've implemented:
1. If/else logic in the flight to select Azure vs. GCP step to handle something
2. Shared logic lives in parent Step class
3. Step classes that extends that parent class with GCP & Azure specific logic 

What about this alternative:
1. No if/else logic in flight - just add IngestCopyLoadHistoryStep
2. In IngestCopyLoadHistoryStep, handle if/else logic for Azure vs. GCP & any shared logic
3. Move any non-shared, Azure or GCP-specific logic into a service
This to me seems easier to understand for readability, but maybe not the recommended way - I'd be interested in discussing why!

Reply:
@nmalfroy See, I'm not the only one who thought this!

Reply:
(@snf2ye The initial implementation of this did exactly as you describe)

Reply:
haha fair enough